File: work/DSA/javascript/recursion_1/firstIndexOfANumber.js
let n = 10
let input = [21, 22, 46, 12, 61, 21, 33, 16, 99, 96]
let x = 25

function firstIndex(input, n, x) {
    if (n == 0) {
        return -1;
    }
    else if (input[0] == x) {
        return 0;
    }
    let subAnswer = firstIndex(input.slice(1), n - 1, x)
    return subAnswer >= 0 ? subAnswer + 1 : subAnswer
}


let answer = firstIndex(input, n, x)
console.log(answer)

File: work/DSA/c++/recursion_1/firstIndexOfANumber.cpp
#include <bits/stdc++.h>
using namespace std;

int firstIndex(int input[], int n, int x)
{
    if (n == 0)
    {
        return -1;
    }
    else if (input[0] == x)
    {
        return 0;
    }
    int ans = firstIndex(input + 1, n - 1, x);
    return ans >= 0 ? ans + 1 : ans;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    int *input = new int[n];

    for (int i = 0; i < n; i++)
    {
        cin >> input[i];
    }

    int x;

    cin >> x;

    cout << firstIndex(input, n, x) << endl;
}